rpcs3/Emu/Cell/Modules/cellSsl.cpp

```diff
@@ -69,9 +69,9 @@ s32 cellSslCertificateLoader(u64 flag, vm::ptr<char> buffer, u32 size, vm::ptr<u
 			final.append(getCert(certPath, i));
 		}
 
-		memset(buffer.get_ptr(), 0, size-1);
-		strncpy(buffer.get_ptr(), final.c_str(), size - 1);
-		buffer.get_ptr()[size - 1] = '\0';
+		memset(buffer.get_ptr(), 0, size);
+		strncpy(buffer.get_ptr(), final.c_str(), size);
+		buffer.get_ptr()[size] = '\0';
```

Comment:
Wait, isn't this guaranteed to write outside the buffer?

Reply:
It doesn't break any games that use it, and TLoU no longer crashes.

Reply:
It's not a proper rationale for introducing a erroneous behaviour/potential UB (though it's is a PS3 buffer that get's written here). It's not even a proper hack.
Unless that's what a real PS3 does it shouldn't be merged into master, especially without a big warning comment. Nobody wants for rpcs3 to turn into a collection of sneaky hacks that will eventually break games as emulation improves.

Reply:
Yes, I agree. I'll write some HW tests soon to confirm the functionality.

Reply:
btw, not sure how helpful the current state of this https://github.com/RPCS3/ps3autotests repo is but there might be a bunch of useful tests (not related to this issue, I think they mainly tests the hw and not sony libs.